[Jeremykhoo] iP

MrJeremyKhoo opened the pull request:
# DukeExtended
duke now can:
`remember to do things - duke `
- store tasks
- remember deadline
- ~~help you be productive~~
all you need is :
1. [time
](https://en.wikipedia.org/wiki/Time)2. patients to **learn**
there are even task lists:
- [ ] like this
- [x]  and this :shipit:


main is 
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

MrJeremyKhoo committed:
finish level-1

MrJeremyKhoo committed:
finish l2

MrJeremyKhoo committed:
done lvl3

MrJeremyKhoo committed:
done level 4

MrJeremyKhoo committed:
finish level-4

MrJeremyKhoo committed:
finish level 6

Junyi00 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+
+	@Override
+	public String toString() {
+		return "[D]" + super.toString() + " (by: " + by + ")";
```
I believe this file has extra level of indentations here weirdly. I notice the same issue occurring in other files too...

Junyi00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+	
+	private static void addList(Task task) {
+		list.add(task);	
+		System.out.println("added: " + task.getDescription());
```
Could it better if we just use the task's string representation here instead? This would consequently remove the need for an extra getter in the Task's class

Junyi00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+	private static void deleteTask(int i) {
+		int index = i - 1;
+		System.out.println("removed: " + list.get(index).toString());
+		System.out.print("You have: " + (list.size() - 1)+ " task(s)");
```
Missing trailing whitespace before '+'. This is a common issue found in other part of the codebase too.

Junyi00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+
+					catch (Exception e) {
+						throw new DukeException(e);	
+					}
```
How about shifting your `try-catch` logic wrapped around the switch-case block instead? That would reduce some code duplication when the same exception could be thrown

Junyi00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+			parseIn(parm);
+		}
+		catch (DukeException e) {
+			
```
I see that the DukeException will send a message to user when it is created, I believe adding a comment for this behaviour would be helpful for others to understand why there is an empty catch block.

Junyi00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+			int byIndex;
+			int fromIndex;
+			String description;
+			List<String> l;
```
`List<String> l` could be given a more descriptive variable name, similar for variable names like `f` and `t`

Junyi00 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+import java.util.Arrays; 
 public class Duke {
-    public static void main(String[] args) {
+	private static ArrayList<Task> list;
```
This is an interesting concept of ensuring only one instance of task list exists. Any reason it is designed this way? If  new Duke was somehow initialised, the array list is 'resetted' base on how the constructor is written now

Junyi00 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,29 @@
+		this.isDone = false;
+	}
+
+	public String getStatusIcon() {
```
This could be set to private since it is only used internally within the class

Junyi00 reviewed (commented):
Great Effort!

MrJeremyKhoo committed:
Level-7

kohkaixun commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+
+	protected String by;
```
There seems to be extra indentation for every line in not just this file, but the other files as well.

kohkaixun commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,211 @@
+					}
+				break;
+			case "event":
+					try{
```
Maybe leave a space between the try and starting curly bracket?

kohkaixun reviewed (commented):
There are probably a few nits to take care of before merging. But those nits are mostly coding standards stuff like more specific names, indentation and spacing.

MrJeremyKhoo committed:
level-8

MrJeremyKhoo committed:
Merge branch 'branch-Level-7'

MrJeremyKhoo committed:
fix bug with event date parsing

MrJeremyKhoo committed:
to merge l8 for auto grader

MrJeremyKhoo committed:
Merge branch 'branch-Level-8'

MrJeremyKhoo committed:
Merge gradle support branch

MrJeremyKhoo committed:
chang tab to 4 space

MrJeremyKhoo committed:
gradle stylecheck

MrJeremyKhoo committed:
add gradle

MrJeremyKhoo committed:
fixed style

MrJeremyKhoo committed:
more oop

MrJeremyKhoo committed:
A-Packages

MrJeremyKhoo committed:
JUnit

MrJeremyKhoo committed:
Revert "more oop"

This reverts commit eef2648893dcf5a2e70df256a5ee40ff71546cf6.

MrJeremyKhoo committed:
Revert "A-Packages"

This reverts commit 57f2ea844553cc4a312e6bf27a731417936db74b.

MrJeremyKhoo committed:
Merge branch 'master' into alt-history

MrJeremyKhoo committed:
add packages

MrJeremyKhoo committed:
java doc

MrJeremyKhoo committed:
add find feature

MrJeremyKhoo committed:
stylecheck graddle

MrJeremyKhoo committed:
merge conflict

MrJeremyKhoo committed:
ready for gradle build

MrJeremyKhoo committed:
before add UI images

MrJeremyKhoo committed:
JFX tutorial p3

MrJeremyKhoo committed:
Level 10

MrJeremyKhoo committed:
Fix exception

MrJeremyKhoo committed:
fix task string update

MrJeremyKhoo committed:
assertion branch

MrJeremyKhoo committed:
Merge pull request #2 from jeremykhoo-NUS/branch-A-Assertions

Branch a assertions

MrJeremyKhoo committed:
bad style

insconsisten trailing whitelines and indentation

ran gradle style check to find those problems

hard to see with eyes, use gradle

MrJeremyKhoo committed:
fix arrowhead indentation

MrJeremyKhoo committed:
Merge pull request #3 from jeremykhoo-NUS/branch-A-CodingQuality

Branch a coding quality

MrJeremyKhoo committed:
to merge for level 10

MrJeremyKhoo committed:
Merge branch 'branch-Level-10'

MrJeremyKhoo committed:
BCD-Extension

MrJeremyKhoo committed:
added UI and UG

MrJeremyKhoo committed:
readme

MrJeremyKhoo committed:
Update README.md
